### DISCUSSION TITLE
Possibly a bug on themes changing automatically and application does not work

### DISCUSSION THREAD

**Discussion by allenmycho0223** [directus/directus#8040] [Archive]
Hi,

I've been seeing several issues multiple times while working with Directus. 
* When I refresh on a page, all of a sudden the background colour (theme) changes from dark to white. I, later on, realised that I can manually set the theme to a specific one and by default, it's set to automatic (or something similar). Hence, I set to the dark theme but it still happens sometimes. I think the solution is to log out and log in again.
* When the theme changes to white mode accidentally, this also usually happens. When I tried to create a new entry in a collection, every field becomes immutable so that I can't do anything.
![image](https://user-images.githubusercontent.com/88812641/133372411-9c15c28f-d516-4d4d-9f35-653002fb0452.png)

Could this be a bug? Also, I'm currently using a specific Directus version `9.0.0-rc.90`. The reason is that we're using several docker images for the application and suddenly some docker image version was updated and it messed up the project.

**Comment by rijkvanzanten** [comment]
@allenmycho0223 Did you run the database migrations when updating the running instances?

**Reply by allenmycho0223** [reply]
@rijkvanzanten No, I only did DB migrations once when generating this Directus project. We're using Azure SQL so I don't know why do I have to run the database migrations. We occasionally make some changes in the Directus and deploy in the Cloud.

**Reply by rijkvanzanten** [reply]
> No, I only did DB migrations once when generating this Directus project. We're using Azure SQL so I don't know why do I have to run the database migrations.

When you update to a newer version of Directus, you should run the database migrations to make sure the internal system tables match the latest version of Directus itself. Seeing you mentioned that "suddenly some docker image version was updated and it messed up the project." I was wondering if you migrated the database when the main Directus version was updated

**Reply by allenmycho0223** [reply]
Sorry, some of my wording might have confused you. Though, for the Directus project, I've been using the version aforementioned and didn't change since then. This is the Dockerfile I'm using. I specified a version so it always uses that version.
![image](https://user-images.githubusercontent.com/88812641/133533983-32df3fff-61e1-4b44-8bb6-6354ad9d18b6.png)

**Reply by rijkvanzanten** [reply]
Ahh gotcha! Thanks for clearing that up. Are you using any caching configuration within Directus? Wondering if it might be the schema cache that needs to be flushed 🤔

**Reply by allenmycho0223** [reply]
I didn't specify any caching related configuration so I suppose they are all set to the default values.

**Reply by rijkvanzanten** [reply]
Are you using any sort of load balanced environment? This is very odd:

> When I refresh on a page, all of a sudden the background colour (theme) changes from dark to white.

🤔 

Trying to figure out what part in the network infrastructure might be causing this issue

**Reply by allenmycho0223** [reply]
The following env variables are being used for the project. FYI, I've deployed the project in Azure AKS cluster. Also, removed some sensitive info
env:
- name: PUBLIC_URL
  value: "/"
- name: DB_CLIENT
  value: "mssql"
- name: DB_HOST
  value: 
- name: DB_PORT
  value: "1433"
- name: DB_DATABASE
  value: 
- name: DB_USER
  value: 
- name: DB_PASSWORD
  value: 
- name: DB_OPTIONS__ENCRYPT
  value: "true"
- name: KEY
  value: 
- name: SECRET
  value: 
- name: EMAIL_FROM
  value: "no-reply@directus.io"
- name: EMAIL_TRANSPORT
  value: "sendmail"
- name: EMAIL_SENDMAIL_NEW_LINE
  value: "unix"
- name: EMAIL_SENDMAIL_PATH
  value: "/usr/sbin/sendmail"
